### DISCUSSION TITLE
Why is `route.id` typed as `string | null`?

### DISCUSSION THREAD

**Discussion by ivanhofer** [sveltejs/kit#9377] [Q&A] [answered]
Isn't it always a `string`? What needs to happen so that `route.id` is `null`?

Context: `ServerLoad` functions

**Comment by wwwdepot** [comment, marked as answer]
route not found when it's null. [From the docs](https://kit.svelte.dev/docs/hooks#shared-hooks-handleerror):

>  in case of a 404 ... you can detect them through `event.route.id` being `null`

**Reply by ivanhofer** [reply]
Ah right, makes sense. Thanks!
